pyi.py

```diff
@@ -97,15 +104,16 @@ class TypeVarInfo(NamedTuple):
     ),
 }
 
-# typing_extensions.ContextManager is omitted from the Y023 and Y027 collections - special-cased
+# Objects you should import from collections.abc/typing instead of typing_extensions
 # We use `None` to signify that the object shouldn't  be parameterised.
+# A Python 2-compatible check
 _BAD_Y023_IMPORTS = {
     # collections.abc aliases
-    "Awaitable": "T",
-    "Coroutine": "YieldType, SendType, ReturnType",
-    "AsyncIterable": "T",
-    "AsyncIterator": "T",
-    "AsyncGenerator": "YieldType, SendType",
+    "Awaitable": _AWAITABLE_SLICE,
+    "Coroutine": _COROUTINE_SLICE,
+    "AsyncIterable": _ITERABLE_SLICE,
+    "AsyncIterator": _ITERATOR_SLICE,
+    "AsyncGenerator": _ASYNCGEN_SLICE,
```

Comment:
Nit: I find most of the `_FOO_SLICE` variables less readable, and I don't see how `_ITERATOR_SLICE` is better than `"T"`. But I like `_COROUTINE_SLICE` for example, because the actual string is complicated.

Reply:
I know what you mean. I did it like that because these all appeared in multiple places, and I tend to be very DRY, even if it makes it slightly less readable. I figured it would make it easier to change the error messages if we ever wanted to in the future.

But I'll make the change :)

Reply:
I've changed all the ones which were just "T".